### ISSUE TITLE
`S.Integers - S.Integers` does not evaluate

### ISSUE THREAD

**Issue by gxyd** [sympy/sympy#10931]
```
>>> S.Integer - S.Integers
Integers() \ Integers()   # should return EmptySet()
>>> S.Integers - S.Reals
Integers() \ (-oo, oo)     # should return EmptySet()
```

**Comment by Nitin216** [comment]
@gxyd can i work on this one.

**Comment by gxyd** [comment]
Yes, you can work on this.

**Comment by Nitin216** [comment]
@gxyd is
``` 
Complement(a,S.Reals)==Complement(a,S.Reals,evaluate=False)
``` 
true or not ??

**Comment by Nitin216** [comment]
ping @gxyd

**Comment by gxyd** [comment]
Sorry, i am not the best person for working on `Sets` module. Probably @smichr or @aktech may be able to help you out for the solution to this issue.

**Comment by nc95** [comment]
@smichr @aktech Could you guide me about this problem.

**Comment by nc95** [comment]
Fixed #10947
